internal/cliwrapper/cliwrapper_test.go

```diff
@@ -50,58 +51,91 @@ func Podman_ImageExists(t *testing.T, connectionName string) {
-	if err := chkDefaultConnectionCmd.Run(); err == nil {
-		Podman_ImageExists(t, connectionName)
-	} else if runtime.GOOS == "linux" {
-		// The podman-machine-default doesn't exist then for Linux, create a temporary socket
-
-		// Setup
-		connectionName = "arcaflow-engine-deployer-podman-test"
-		podmanSocketPath := "unix:///var/tmp/" + connectionName + ".sock"
-
-		tmpPodmanSocketCmd = exec.Command(tests.GetPodmanPath(), "system", "service", "--time=0", podmanSocketPath)
-		if err := tmpPodmanSocketCmd.Start(); err != nil {
-			t.Fatalf("Failed to create temporary podman socket")
+	chkDefaultConnectionCmd := exec.Command(tests.GetPodmanPath(), "--connection", connectionName, "system", "info") //nolint:gosec  // Command line is trusted
+	if err := chkDefaultConnectionCmd.Run(); err != nil {
+		// The podman-machine-default connection doesn't exist, so try to create
+		// an alternative connection service.  For now, only try this on Linux.
+		//
+		//goland:noinspection GoBoolExpressions  // The linter cannot tell that this expression is not constant.
+		if runtime.GOOS != "linux" {
+			t.Skipf("There is no default Podman connection and no support for creating it on %s.", runtime.GOOS)
 		}
 
-		addConnectionCmd := exec.Command(tests.GetPodmanPath(), "system", "connection", "add", connectionName, podmanSocketPath)
-		if err := addConnectionCmd.Run(); err != nil {
-			t.Fatalf("Failed to add connection: " + connectionName)
+		connectionName = createPodmanConnection(t)
+	}
+
+	// Run the test
+	podmanImageExists(t, &connectionName)
+}
+
+// createPodmanConnection creates a Podman API service process and configures
+// a Podman "connection" to allow it to be used for remote Podman invocations.
+func createPodmanConnection(t *testing.T) (connectionName string) {
+	// Setup:  create a temporary directory with a random name, to avoid
+	// collisions with other concurrently-running tests; use the resulting
+	// path as the name of the Podman service connection and put the service
+	// socket in the directory.  Start a listener on that socket and configure
+	// a connection to it.  Declare cleanup functions which will remove the
+	// connection, kill the listener, and remove the temporary directory and
+	// socket.
+	t.Logf("Adding a local Podman API service and connection.")
+	sockDir, err := os.MkdirTemp("", "arcaflow-engine-deployer-podman-test-*")
+	if err != nil {
+		t.Fatalf("Unable to create socket directory: %q", err)
+	}
+
+	t.Cleanup(func() {
+		t.Logf("Removing socket directory, %q.", sockDir)
+		if err := os.RemoveAll(sockDir); err != nil {
+			t.Logf("Unable to remove socket directory, %q: %q", sockDir, err)
 		}
+	})
 
-		// Run test
-		Podman_ImageExists(t, connectionName)
+	t.Logf("Local Podman API service connection is %q.", sockDir)
 
-		// Clean up
-		if err := tmpPodmanSocketCmd.Process.Kill(); err != nil {
-			t.Fatalf("Failed to kill temporary socket")
+	connectionName = sockDir
```

Comment:
This should return something like `${TMPDIR}/arcaflow-engine-deployer-podman-test-<nnn>`. Is that really what you want to use for the connection *name*? (Does podman even guarantee it accepts slashes in the connection name?) I'd think you'd want `filepath.Base(filepath.Dir(sockDir))`...

(Although ... `podman` does seem to accept slashes in the name, so I guess it'll work. It still feels odd, but I suppose there's no need to overcomplicate this...)

Reply:
I checked that Podman would accept the slashes, and then I checked that the test worked, so I so I figured it was good enough, especially since, presumably, no one but the test itself would ever see or use the connection name.  (And, it was _way_ simpler than parsing out the directory base name.  🙂)